Library/brew.sh

```diff
@@ -1,16 +1,46 @@
-odie() {
-  if [[ -t 2 ]] # check whether stderr is a tty.
+styled-output() {
+  local message output_fd="$1" plain="$2" styled="$3"
+  shift 3
+
+  if [[ $# -eq 0 ]]
   then
-    echo -ne "\033[4;31mError\033[0m: " >&2 # highlight Error with underline and red color
+    IFS= read -d '' -r message # Read message from `stdin`.
+    message="${message%$'\n'}" # Trim trailing newline.
   else
-    echo -n "Error: " >&2
+    message="$*"
   fi
-  if [[ $# -eq 0 ]]
+
+  if [[ -t "$output_fd" ]] # If FD is a TTY, style message with color/bold/...
   then
-    /bin/cat >&2
+    printf "$styled\n" "$message"
   else
-    echo "$*" >&2
+    printf "$plain\n" "$message"
   fi
+}
+
+ohai() {
+  # Use bold and blue color if `stdout` is a TTY.
+  styled-output 1 "==> %s" "\033[1;34m==>\033[1;39m %s\033[0m" "$@"
+}
+
+oh1() {
+  # Use bold and green color if `stdout` is a TTY.
+  styled-output 1 "==> %s" "\033[1;32m==>\033[1;39m %s\033[0m" "$@"
+}
+
+opoo() {
+  # Use underline and yellow color if `stderr` is a TTY.
+  styled-output 2 "Warning: %s" "\033[4;33mWarning\033[0m: %s" "$@" >&2
+}
+
+onoe() {
+  # Use underline and red color if `stderr` is a TTY.
+  styled-output 2 "Error: %s" "\033[4;31mError\033[0m: %s" "$@" >&2
+}
+
+odie() {
+  # Use underline and red color if `stderr` is a TTY.
+  styled-output 2 "Error: %s" "\033[4;31mError\033[0m: %s" "$@" >&2
```

Comment:
Yes, that makes a lot of sense! I just wanted to get them implemented and reviewed first before worrying about placement. I'm also not entirely sure if they should live in `Library/Homebrew` or somewhere else? It feels both weird and natural to put them next to their Ruby counterparts.